server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -68,28 +44,27 @@ func (ds *cachedMysql) NewAppConfig(ctx context.Context, info *fleet.AppConfig)
 		return nil, err
 	}
 
-	err = ds.storeInRedis(CacheKeyAppConfig, ac)
+	ds.mu.Lock()
+	ds.ac = ac
+	ds.acLastErr = nil
+	ds.mu.Unlock()
 
-	return ac, err
+	return ac, nil
 }
 
 func (ds *cachedMysql) AppConfig(ctx context.Context) (*fleet.AppConfig, error) {
```

Comment:
Only worried that if a developer uses the result of `AppConfig` (`*fleet.AppConfig`) and modifies it, then for a second that result could end up on the cache (because it's a pointer).

Reply:
Ah yeah good point. I think the call on line 57 needs to copy the value rather than the pointer as otherwise there's still a race condition if we're only copying and returning the pointer.